app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -42,6 +43,11 @@ protected void onCreate(Bundle savedInstanceState) {
             setupDrawerContent(navigationView);
         }
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            getWindow().setStatusBarColor(getResources().getColor(R.color.color_primary));
```

Comment:
That's a pretty broad issue in itself, don't think setting programmatically
here makes a difference because the rest of the palette is immutable once
declared in XML, there isn't a way to change the theme, the textView
cursors etc all derive from the palette colours defined in XML. Even if we
are able to do this, we still then need to manually change accent colours
for ALL the widgets because they will look out of place. I would recommend
placing the theme colours inside XML at this stage.

But feel free to work this out in a separate issue. Status bar colors are
only 1 spec of the material theme, there is widget tinting, highlightColors
and a whole bunch of other properties that need addressing. It would look
odd if the theme colors are red, the status bar is blue and the cursor or
the switch, or the checkbox is blue in color.

The XML already allows a declarative style semantic, so I would in my
opinion just stick with it.

On Wed, Jul 1, 2015 at 1:17 AM, Arnav Gupta <notifications@github.com>
wrote:

> In app/src/main/java/org/fossasia/openevent/activities/MainActivity.java
> <https://github.com/fossasia/open-event-android/pull/47#discussion_r33598346>
> :
>
> > @@ -42,6 +43,11 @@ protected void onCreate(Bundle savedInstanceState) {
> >              setupDrawerContent(navigationView);
> >          }
> >
> > +        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
> > +            getWindow().setStatusBarColor(getResources().getColor(R.color.color_primary));
>
> Can be done from the server. Set a primary color field in the event
> endpoint. Fetch it, and setColor programmatically wherever required?
> … <#14e45797cda6dfd3_>
> On Jun 30, 2015 10:43 PM, "Mohit Kanwal" <notifications@github.com>
> wrote: In
> app/src/main/java/org/fossasia/openevent/activities/MainActivity.java <
> https://github.com/fossasia/open-event-android/pull/47#discussion_r33597794>
> : > @@ -42,6 +43,11 @@ protected void onCreate(Bundle savedInstanceState) {
> > setupDrawerContent(navigationView); > } > > + if (Build.VERSION.SDK_INT
> >= Build.VERSION_CODES.LOLLIPOP) { > +
> getWindow().setStatusBarColor(getResources().getColor(R.color.color_primary));
> Would depend on the event I guess. Or manually customisable by changing a
> themes.xml for different events. Or we a script to write it altogether. —
> Reply to this email directly or view it on GitHub <
> https://github.com/fossasia/open-event-android/pull/47/files#r33597794>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/open-event-android/pull/47/files#r33598346>.
>



-- 
Mohit Singh KANWAL (Mr)

[image: Facebook] <http://facebook.com/mohitkanwal>  [image: Twitter]
<http://twitter.com/mohitkanwal>